app/models/submission.rb

```diff
@@ -15,7 +15,28 @@ class Submission < ApplicationRecord
   has_many :submission_packets, dependent: :destroy
   has_many :poems, through: :submission_packets
 
+  accepts_nested_attributes_for :submission_packets,
+    reject_if: :empty_packet?
+
   validates :title, presence: true
-  validates :status, presence: true, inclusion: { in: SUBMISSION_STATUSES }
+  validates :status, presence: true, inclusion: { in: STATUSES }
   validates :submitted_to, presence: true
+
+  def empty_packet?(attributes)
+    attributes[:poem_attributes].each_value.all?(&:blank?)
+  end
+
+  def deep_blank?(attributes)
+    attributes.each_pair do |key, value|
+      binding.pry
+      if value.respond_to?(:keys)
+        deep_blank?(value)
+      elsif value.blank?
+        true
+      else
+        false
```

Comment:
this is one place that you should probably return early. Might be something helpful to TDD together.